frontend/src/pages/Mesh/target/TargetPanelControlPlane.tsx

```diff
@@ -39,14 +36,22 @@ import { load, dump } from 'js-yaml';
 import { yamlDumpOptions } from '../../../types/IstioConfigDetails';
 import { CertsInfo } from 'types/CertsInfo';
 import { IstioCertsInfo } from 'components/IstioCertsInfo/IstioCertsInfo';
+import { namespaceItemsSelector } from 'store/Selectors';
+import { Namespace } from 'types/Namespace';
+import { connect } from 'react-redux';
+import { KialiAppState } from '../../../store/Store';
 
-type TargetPanelControlPlaneProps = TargetPanelCommonProps & {
-  meshStatus: string;
-  minTLS: string;
+type ReduxProps = {
+  namespaces: Namespace[];
```

Comment:
It does actually stay in sync but somewhat tangentially. There's a component in the masthead (`IstioStatus`) that is hooked into the refresh button so when a new namespace is added with the target namespace panel open and you hit the refresh button or the refresh timer ticks, the `IstioStatus` component will update the redux namespace store causing this panel to be updated.

The namespace count that you see on the dataplane node comes from the graph info so the two can get temporarily out of sync. Since this component is tied in with the graph, the namespace info should probably come from the graph also. I'll either plumb it through as props or pull it from the graph response and add it to the redux store.

Reply:
OK, that's interesting.  The Namespace Dropdown has to be up to date, so in that case we ensure an update outside of the standard refresh/status plumbing.  The graph updates at refresh time, so the graph should basically be in sync with refresh, so not sure there needs to be any further sync work but do what you think is needed to ensure namespace changes are reflected.

Reply:
Alright the namespaces actually come from the graph response. They are included in the `infraData` of the controlplane nodes. When the graph request returns, the frontend saves the controlplanes in the redux state. This way the namespaces that get displayed on the graph will always stay in sync with the namespaces on the target panel.